### PR TITLE
Replace tel/email/urls in string, by HTML tags 

### PR DESCRIPTION
**Pull request by edellucien**

### CODE REVIEW

File: src/popups/HandicapPopup/HandicapPopupElement.js [hunk 1]
```diff
@@ -8,6 +8,121 @@ const propTypes = {
...
+  });
+  // Flatten array of matches.
+  if (matched.length) {
+    matched = matched.reduce((acc, curr) => [...acc, ...curr], []);
+  }
+
+  if (matched) {
+    // Remove duplicates from match array.
+    return matched.sort().filter((item, pos, ary) => {
+      return !pos || item !== ary[pos - 1];
```
Comment (oterral) [inline src/popups/HandicapPopup/HandicapPopupElement.js]: ary.indexOf(item) === pos
```suggestion
      return ary.indexOf(item) === pos;
```

File: src/popups/HandicapPopup/HandicapPopupElement.js [hunk 2]
```diff
@@ -8,6 +8,121 @@ const propTypes = {
...
+
+const replaceLinks = (intialTextArray, matched, renderCallback) => {
+  // Construct an RegExp to capture all matched emails.
+  const regularExp = matched.reduce(
+    (prevVal, currVal, currentIndex) =>
+      currentIndex === 0 ? prevVal : `${prevVal}|(${escapeRegExp(currVal)})`,
+    `(${escapeRegExp(matched[0])})`,
+  );
+
+  // Split array items and flatten it.
```
Comment (oterral) [inline src/popups/HandicapPopup/HandicapPopupElement.js]: array.flat() exists. Could it be useful ?

File: src/popups/HandicapPopup/HandicapPopupElement.js [hunk 3]
```diff
@@ -8,6 +8,121 @@ const propTypes = {
...
+  const flatSplitArray = intialTextArray
+    .map(item => {
+      if (typeof item === 'string') {
+        return item
+          .split(new RegExp(regularExp, 'g'))
+          .filter(v => v !== undefined && v !== '');
```
Comment (oterral) [inline src/popups/HandicapPopup/HandicapPopupElement.js]: Describe in a comment what's going on there.

File: src/popups/HandicapPopup/HandicapPopupElement.js [hunk 4]
```diff
@@ -8,6 +8,121 @@ const propTypes = {
...
+  let matched = [];
+  testers.forEach(t => {
+    const matches = intialText.match(t);
+    if (matches) {
+      matched.push(matches);
+    }
+  });
+  // Flatten array of matches.
+  if (matched.length) {
+    matched = matched.reduce((acc, curr) => [...acc, ...curr], []);
```
Comment (oterral) [inline src/popups/HandicapPopup/HandicapPopupElement.js]: matched.flat()

File: src/popups/HandicapPopup/HandicapPopupElement.js [hunk 5]
```diff
@@ -8,6 +8,121 @@ const propTypes = {
...
+const emailTester = /[a-zA-Z0-9._+%-]+@[a-zA-Z0-9.-]+[.][a-zA-Z]{2,4}/gm;
+const urlTester = /(www)?[a-z0-9]+([\\-\\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?/g;
+
+const escapeRegExp = string => {
+  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
+};
+
+const searchMatches = (intialText, testers) => {
+  let matched = [];
+  testers.forEach(t => {
```
Comment (oterral) [inline src/popups/HandicapPopup/HandicapPopupElement.js]: let matched = [...tester.filter((t)=> intialText.match(t))]

File: src/popups/HandicapPopup/HandicapPopupElement.js (hunk 4 above)
Comment (edellucien) [reply src/popups/HandicapPopup/HandicapPopupElement.js]: Not supported on our beloved browsers: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat

File: src/popups/HandicapPopup/HandicapPopupElement.js (hunk 2 above)
Comment (edellucien) [reply src/popups/HandicapPopup/HandicapPopupElement.js]: not supported on ie/edge

File: src/popups/HandicapPopup/HandicapPopupElement.js (hunk 3 above)
Comment (oterral) [reply src/popups/HandicapPopup/HandicapPopupElement.js]: I guess 
```suggestion
          .filter(v => !!v');
```
does the same thing.

File: src/popups/HandicapPopup/HandicapPopupElement.js (hunk 4 above)
Comment (oterral) [reply src/popups/HandicapPopup/HandicapPopupElement.js]: Test if it's polyfill automatically on IE

**Comment by edellucien** [comment]
Mario decided to keep this here for now on (see ticket).

**Comment by oterral** [comment]
I will not approve this without tests ;)

**Comment by edellucien** [comment]
Indeed! Tests are ready for review :smile:

File: src/popups/HandicapPopup/HandicapPopupElement.js (hunk 4 above)
Comment (edellucien) [reply src/popups/HandicapPopup/HandicapPopupElement.js]: It works you are right :)

File: src/popups/HandicapPopup/HandicapPopupElement.js (hunk 2 above)
Comment (edellucien) [reply src/popups/HandicapPopup/HandicapPopupElement.js]: but polyfilled in this app, you are right! :+1:

File: src/popups/HandicapPopup/HandicapPopupElement.js [hunk 6]
```diff
@@ -8,6 +8,120 @@ const propTypes = {
...
+  );
+
+  // Split array items and flatten it.
+  const flatSplitArray = intialTextArray
+    .map(item => {
+      if (typeof item === 'string') {
+        // Split string with RegExp and remove all empty strings or undefined.
+        return item
+          .split(new RegExp(regularExp, 'g'))
+          .filter(v => !!v');
```
Comment (edellucien) [inline src/popups/HandicapPopup/HandicapPopupElement.js]: I accepted your suggestion, but there was a apostrophe too much
```suggestion
          .filter(v => !!v);
```

**Comment by oterral** [comment]
@edellucien fix the build then merge

**Comment by edellucien** [comment]
Flat method is problematic in node v10 for the tests. I check it

File: src/popups/HandicapPopup/HandicapPopupElement.test.js [hunk 7]
```diff
@@ -3,6 +3,7 @@ import { configure, mount } from 'enzyme';
 import Adapter from 'enzyme-adapter-react-16';
 import renderer from 'react-test-renderer';
 import HandicapPopupElement from './HandicapPopupElement';
+import '../../utils/polyfills/flat';
```
Comment (oterral) [inline src/popups/HandicapPopup/HandicapPopupElement.test.js]: Can't you add react-app-polyfill/ie11'  and 'react-app-polyfill/stable' instead ?

Comment (edellucien) [reply src/popups/HandicapPopup/HandicapPopupElement.test.js]: Even better you are right :)

**Pull request merged**